File: PRNumberBiz/app/src/main/java/com/pplus/prnumberbiz/core/network/prnumber/IPRNumberRequest.java
package com.pplus.prnumberbiz.core.network.prnumber;

import com.pplus.prnumberbiz.core.network.model.response.NewResultResponse;

import network.common.PplusCallback;

/**
 * Created by j2n on 2016. 9. 12..
 */
public interface IPRNumberRequest<T> {

    IPRNumberRequest<T> setCallback(PplusCallback<NewResultResponse<T>> callback);

    IPRNumberRequest<T> setTag(Object tag);

    PRNumberRequest<T> build();
}

File: LuckyBol/app/src/main/java/com/pplus/luckybol/core/network/prnumber/IPRNumberRequest.kt
package com.pplus.luckybol.core.network.prnumber

import com.pplus.networks.common.PplusCallback
import com.pplus.luckybol.core.network.model.response.NewResultResponse

/**
 * Created by j2n on 2016. 9. 12..
 */
interface IPRNumberRequest<T> {
    fun setCallback(callback: PplusCallback<NewResultResponse<T>>): IPRNumberRequest<T>
    fun setTag(tag: Any): IPRNumberRequest<T>
    fun build(): PRNumberRequest<T>
}